docs/source/essentials/schema.md

```diff
@@ -17,55 +30,212 @@ type Book {
+
+Thanks to the relationship between `Book` and `Author`, which is defined in the SDL above, this `query` would look like:
+
+```graphql
+query {
+  getBooks {
+    title
+    author {
+      name
+    }
+  }
+}
+```
+
+And, without additional effort on its part, the client would receive the information in the same shape as the request:
+
+```json
+{
+  "data": {
+    "getBooks": [
+      {
+        "title": "..."
+        "author": {
+          "name": "..."
+        }
+      },
+      ...
+    ]
+  }
+}
+```
+
+### Query arguments
+
+> TODO? Should this section exist here?
 
 ## Mutations
 
-Mutations are operations sent to the server to create, update or delete data.  Those familiar with REST-based communication verbs would associate these with the `PUT`, `POST`, `PATCH` and `DELETE` methods.
+Mutations are operations sent to the server to create, update or delete data.  These are comparable to the `PUT`, `POST`, `PATCH` and `DELETE` verbs on REST-based APIs.
 
-* Mutations are the main entry point to make updates to the data backing Queries
-* Here is what one looks like coming from the client
-* Here is the corresponding server schema
-* A mutation can contain arguments and fields
-* Input types are passed to Apollo Server and inform it how to make the update
-* The fields describe the object that is returned by a mutation, often times the object that was created or the entire collection that was modified. Or a confirmation of deletion
-  * Here is what the return looks like from the previous mutation call
+Much like how the `Query` type defines the entry-points for data-fetching operations on a GraphQL server, the root-level `Mutation` type specifies the entry points for data-manipulation operations.
 
-## Introspection
+For example, when imagining a situation where the API supported adding a new `Book`, the SDL might implement the following `Mutation` type:
+
+```graphql
+type Mutation {
+  addBook(title: String, author: String): Book
+}
+```
+
+This implements a single `addBook` mutation which accepts a `title` attribute (a `String`) and returns the newly-created `Book` object.
+
+The new `Book` object matches the previously-created `Book` type and, much like the `Query` type, the fields we desire to receive back from the operation are specified when sending the `mutation`:
+
+```graphql
+mutation {
+  addBook(title: "Fox in Socks", author: "Dr. Seuss") {
+    title
+    author {
+      name
+    }
+  }
+}
+```
+
+The result of this mutation would be:
+
+```json
+{
+  "data": {
+    "addBook": {
+      {
+        "title": "Fox in Socks",
+        "author": {
+          "name": "Dr. Seuss"
+        }
+      }
+    }
+  }
+}
+```
+
+Multiple mutations may be sent in the same request, however they will be executed in the order they are provided (in series), in order to avoid race-conditions within the operation.
```

Comment:
Multiple mutations could refer to multiple top level mutations that need to be differentiated by an OperationName. I think this means root mutation fields. An example to illustrate might help?

```graphql
mutation {
  addBook(title: "Fox in Socks", author: "Dr. Seuss") {
   title
    author {
      name
    }
 }
 removeBook(title: "Fox in Socks", author: "Dr. Seuss") {
    ...
  }
}
```
vs the other way around

Reply:
@evans Are they executed differently?

Reply:
@abernix 

```graphql
mutation {
  addBook(title: "Fox in Socks", author: "Dr. Seuss") {
   ...
  }
 removeBook(title: "Fox in Socks", author: "Dr. Seuss") {
    ...
  }
}

mutation {
 removeBook(title: "Fox in Socks", author: "Dr. Seuss") {
    ...
  }
  addBook(title: "Fox in Socks", author: "Dr. Seuss") {
   ...
  }
}
```

With my understanding of how things are run serially, the first request would end up with no book added and the second would return an error and add the book.

The thought came into my mind that the scenario above could be confused with something like, where the operation name is `AddBook` or `RemoveBook` and only one operation is run: 

```graphql
mutation AddBook {
  addBook(title: "Fox in Socks", author: "Dr. Seuss") {
   ...
  }
}

mutation RemoveBook {
 removeBook(title: "Fox in Socks", author: "Dr. Seuss") {
    ...
  }
}
```